### PR TITLE
Check that points satisfy the curve equation

### PR DESCRIPTION
**Pull request by emillon**
(See #3)

This checks that points parsed from `Fe.of_cstruct` satisfy the curve equation. It enforces a nice invariant: values typed `Point.t` represent the point at infinity or a point on the curve.

### CODE REVIEW

File: p256/point.ml [hunk 1]
```diff
@@ -9,26 +9,103 @@ let at_infinity () =
   let f_z = Fe.create () in
   {f_x; f_y; f_z}
 
+let fe_from_be_cstruct cs =
+  let cs_rev = Cstruct.rev cs in
+  let fe = Fe.create () in
+  Fe.from_bytes fe cs_rev; Fe.to_montgomery fe; fe
+
+let is_on_curve ~x ~y =
+  let a =
```
Comment (emillon) [inline p256/point.ml]: should we extract a `Parameters` module?

**Comment by emillon** [comment]
cc @pascutto

File: p256/point.ml (hunk 1 above)
Comment (pascutto) [reply p256/point.ml]: That might be a good idea since we'll probably need them elsewhere too (eg the curve order for #2)

Comment (emillon) [reply p256/point.ml]: done (I put the generator there as well)

**Pull request merged**